library/common/config/config.cc

```diff
@@ -97,6 +97,184 @@ const char* socket_tag_config_insert = R"(
+        keepalive_interval: 5
+        keepalive_probes: 1
+        keepalive_time: 10
+    circuit_breakers: &circuit_breakers_settings
+      thresholds:
+      - priority: DEFAULT
+        # Don't impose limits on concurrent retries.
+        retry_budget:
+          budget_percent:
+            value: 100
+          min_retry_concurrency: 0xffffffff # uint32 max
+      per_host_thresholds:
+      - priority: DEFAULT
+        max_connections: *max_connections_per_host
+    typed_extension_protocol_options: *alpn_protocol_options
+  - name: base_clear
+    connect_timeout: *connect_timeout
+    lb_policy: CLUSTER_PROVIDED
+    cluster_type: *base_cluster_type
+    transport_socket:
+      name: envoy.transport_sockets.http_11_proxy
+      typed_config:
+        "@type": type.googleapis.com/envoy.extensions.transport_sockets.http_11_proxy.v3.Http11ProxyUpstreamTransport
+        transport_socket:
+          name: envoy.transport_sockets.raw_buffer
+          typed_config:
+            "@type": type.googleapis.com/envoy.extensions.transport_sockets.raw_buffer.v3.RawBuffer
+    upstream_connection_options: *upstream_opts
+    circuit_breakers: *circuit_breakers_settings
+    typed_extension_protocol_options: *h1_protocol_options
+  - name: base_h2
+    typed_extension_protocol_options:
+      envoy.extensions.upstreams.http.v3.HttpProtocolOptions:
+        "@type": type.googleapis.com/envoy.extensions.upstreams.http.v3.HttpProtocolOptions
+        explicit_http_config:
+          http2_protocol_options: {}
+    connect_timeout: *connect_timeout
+    lb_policy: CLUSTER_PROVIDED
+    cluster_type: *base_cluster_type
+    transport_socket: *base_h2_socket
+    upstream_connection_options: *upstream_opts
+    circuit_breakers: *circuit_breakers_settings
+    typed_extension_protocol_options: *h2_protocol_options
+  - name: base_h3
+    connect_timeout: *connect_timeout
+    lb_policy: CLUSTER_PROVIDED
+    cluster_type: *base_cluster_type
+    transport_socket: *base_h3_socket
+    upstream_connection_options: *upstream_opts
+    circuit_breakers: *circuit_breakers_settings
+    typed_extension_protocol_options: *h3_protocol_options
+)";
+
+const char* transport_socket_insert = R"(
+      transport_socket:
+        name: envoy.transport_sockets.tls
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext
+          common_tls_context:
+            tls_certificate_sds_secret_configs:
+              - name: client_cert
+                sds_config:
+                  api_config_source:
+                    api_type: GRPC
+                    grpc_services:
+                      - envoy_grpc:
+                          cluster_name: xds_cluster.lyft.com
+                        timeout: 300s
+                    transport_api_version: V3
+                  resource_api_version: V3
+          sni: lyft.com
+)";
+
+const char* rtds_clusters_insert = R"(
+    - name: base_h2
```

Comment:
base_h2 is not an xDS cluster, is it?  and it's already declared in the default clusters in L197 above

Reply:
Talked offline: we may want to have this yaml get passed in to the function instead.